Upgrade default GHC version

Hi,

I just tested today the following:
```shell
$ git clone --recursive git@gitlab.haskell.org:ghc/ghc.git
$ cd ghc/
$ git clone https://github.com/alpmestan/ghc.nix
$ nix-shell ghc.nix
$ ./boot
$ ./configure 
```

The last step (`./configure`) fails as such:
```
checking for gfind... no
checking for find... /nix/store/xhdbfv3kivss6y04xk9i4a5pknaww1az-findutils-4.7.0/bin/find
checking for sort... /nix/store/3kqc2wmvf1jkqb2jmcm7rvd9lf4345ra-coreutils-8.31/bin/sort
checking for GHC version date... inferred 8.11.0.20200331
checking for GHC Git commit id... inferred 4b9c586472bf99425f7bbcf346472d7c54f05028
checking for ghc... /nix/store/hcrblsvvbmamb2lwsv68n87dvncr6pq3-ghc-8.6.5/bin/ghc
checking version of ghc... 8.6.5
configure: error: GHC version 8.8 or later is required to compile GHC.
```

I think the default version of GHC should be upgraded to GHC 8.8 or more, what do you think?

Cheers!

Note: This seems to require an update of `nixpkgs.src-json` which is done using `nix/update-pin.sh nixpkgs HEAD` or `nix/update-pin.sh nixpkgs be0d1dc5a6c42456dbddde9b347d49cfaad034ef` if I'm not mistaken?

--- a/README.md
+++ b/README.md
@@ -48,8 +48,9 @@ $ cabal update
 ```
 
 
-## Using `ghcide`
+## Using `ghcide` (⚠️ currently unavailable)
 
+<del>
 You can also use `ghc.nix` to provide the right version of `ghcide` if you
 want to use `ghcide` whilst developing on GHC. In order to do so, pass the `withIde`
 argument to your `nix-shell` invocation.
@@ -57,6 +58,12 @@ argument to your `nix-shell` invocation.
 ```
 nix-shell ~/.ghc.nix --arg withIde true
 ```
+</del>
+GHCIDE support is currently unavailable for GHC > 8.8.
+
+See:
+- https://github.com/cachix/ghcide-nix/issues/3
+- https://github.com/alpmestan/ghc.nix/issues/64
 
 ## Running `./validate`
 

--- a/default.nix
+++ b/default.nix
@@ -10,14 +10,17 @@ let
 in
 { nixpkgsPin ? ./nix/pins/nixpkgs.src-json
 , nixpkgs   ? import (fetchNixpkgs nixpkgsPin) {}
-, bootghc   ? "ghc865"
+, bootghc   ? "ghc882"
 , version   ? "8.9"
 , hadrianCabal ? (builtins.getEnv "PWD") + "/hadrian/hadrian.cabal"
 , useClang  ? false  # use Clang for C compilation
 , withLlvm  ? false
 , withDocs  ? true
 , withGhcid ? false
-, withIde   ? false
+# GHCIDE support on hold as we must use GHC 8.8 minimum for GHC development, and GHCIDE is not yet available for GHC 8.8
+# See https://github.com/cachix/ghcide-nix/issues/3
+# See https://github.com/alpmestan/ghc.nix/issues/64
+# , withIde   ? false
 , withHadrianDeps ? false
 , withDwarf  ? nixpkgs.stdenv.isLinux  # enable libdw unwinding support
 , withNuma   ? nixpkgs.stdenv.isLinux
@@ -69,7 +72,7 @@ let
       ++ optional withNuma numactl
       ++ optional withDwarf elfutils
       ++ optional withGhcid ghcid
-      ++ optionals withIde [ghcide ccls bear]
+      # ++ optionals withIde [ghcide ccls bear]
       ++ optional withDtrace linuxPackages.systemtap
       ++ (if (! stdenv.isDarwin)
           then [ pxz ]

--- a/nix/pins/nixpkgs.src-json
+++ b/nix/pins/nixpkgs.src-json
@@ -1,6 +1,6 @@
 {
   "url": "https://github.com/NixOS/nixpkgs",
-  "rev": "c5aabb0d603e2c1ea05f5a93b3be82437f5ebf31",
-  "sha256": "15fwszhn6078sbrb8qk83g8afvh4qnmvff0qbkbvq3cm1fxni2w1",
+  "rev": "6d445f8398d2d585d20d9acacf00fd9d15081b3b",
+  "sha256": "1ajd0zr31iny3g9hp0pc1y2pxcm3nakdv6260lnvyn0k4vygync2",
   "fetchSubmodules": false
 }

